include/switcher.php

```diff
@@ -4,11 +4,28 @@
  */
 
 /**
- * A class to display a language switcher on frontend
+ * A class to display a language switcher
  *
  * @since 1.2
  */
-class PLL_Switcher {
+abstract class PLL_Switcher {
+
+	/**
+	 * Define which class of the switcher to return.
+	 *
+	 * @param PLL_Base $polylang Polylang Object.
+	 *
+	 * @return PLL_Admin_Switcher|PLL_Frontend_Switcher|null
+	 */
+	public static function create( $polylang ) {
+		if ( $polylang instanceof PLL_Frontend ) {
+			return new PLL_Frontend_Switcher();
+		} elseif ( $polylang instanceof PLL_Admin ) {
+			return new PLL_Admin_Switcher();
+		} else {
+			return null;
+		}
+	}
```

Comment:
Couldn't we only pass the $links object and test its kind of instance instead of the Polylang object ?

Reply:
Isn't there a REST case where we need the Front-End Switcher? Are the links also an instance of `PLL_Admin_Links` in this case?

Reply:
Yes! In REST context we are using `PLL_Admin_Links`. We manage the detection by the use of a `is_block_editor` parameter passed to the request.
Otherwise we decided all together to keep passing the Polylang full object.